src/identity/configuration/IdentityProviderFactory.ts

```diff
@@ -389,13 +391,17 @@ export class IdentityProviderFactory implements ProviderFactory {
 
       this.logger.warn(`OIDC request failed: ${detailedError}`);
 
+      let resultingError: HttpError = new OAuthHttpError(out, error.name, oidcError.statusCode, error.message);
+      // Keep the original stack to make debugging easier
+      resultingError.stack = error.stack;
```

Comment:
This is usually done by `detailedError.cause = originalError`.

…but I see we have 7 error objects here already so not sure which is which anymore 😅

Reply:
I wasn't sure about copying the stack or not. It also doesn't matter that much. Only impact it has is if you start the server with stack tracing enabled that this will be shown on the error pages. But in practice the impact is minimal.

The reason for not passing the OIDC error object along is to not have a reference to that type in our error object, as that type belongs to the OIDC library (specifically the fact that it has a `statusCode` field).

Also `error` and `oidcError` are the same object, except the second one is cast to the OIDC error type. I might have been to hastility there in not reusing the same variable name for every entry 😅.